### PR TITLE
Take care of the worst compile-time offenders

### PR DESCRIPTION
**Pull request by UplinkCoder**
Reduces the memory and time taken by around 50%

### CODE REVIEW

File: test/tools/d_do_test.d [hunk 1]
```diff
@@ -607,14 +607,19 @@ string unifyNewLine(string str)
     // On Windows, Outbuffer.writenl() puts `\r\n` into the buffer,
     // then fprintf() adds another `\r` when formatting the message.
     // This is why there's a match for `\r\r\n` in this regex.
-    static re = regex(`\r\r\n|\r\n|\r|\n`, "g");
-    return std.regex.replace(str, re, "\n");
+
+    static Regex!char* re;
+    if (re !is null)
```
Comment (kubo39) [inline test/tools/d_do_test.d:612]: opposite?

**Comment by MoonlightSentinel** [comment]
Well, the time and memory requirement certainly isn't pretty. Did you measure the impact on the test suite runtime (because it's now recompiling those regexes for every test)?

File: test/tools/d_do_test.d (hunk 1 above)
Comment (UplinkCoder) [reply test/tools/d_do_test.d:612]: yes. doh!

**Comment by UplinkCoder** [comment]
@MoonlightSentinel I did not.
We will seen once it runs this test suite.

**Comment by UplinkCoder** [comment]
huh it does work ... well back to the drawing board.
@MoonlightSentinel Now that you know regex having this impact could you try rewriting the code to not use regex?

**Comment by MoonlightSentinel** [comment]
Well, the windows/non-windows path handling is on my list but i don't have an ETA for that.

**Pull request closed without merging**